src/util.js

```diff
@@ -47,16 +51,45 @@ export function pathRankSort(a, b) {
 		bAttr = b.attributes || EMPTY;
 	if (aAttr.default) return 1;
 	if (bAttr.default) return -1;
-	let diff = rank(aAttr.path) - rank(bAttr.path);
-	return diff || (aAttr.path.length - bAttr.path.length);
+	let aRank = rank(aAttr.path),
+		bRank = rank(bAttr.path);
+	return (aRank < bRank) ? 1 :
+		(aRank == bRank) ? 0 :
+		-1;
 }
 
 export function segmentize(url) {
 	return strip(url).split('/');
 }
 
-export function rank(url) {
-	return (strip(url).match(/\/+/g) || '').length;
+export function rank(path) {
+	return strip(path).
+	        replace(/(:)?([^\/]*?)([*+?])?(?:\/+|$)/g, (match, isParam, segment, flag) => {
```

Comment:
I just re-read your comment about the ranking and take back the numeric bit. Still nice to have the exec()-based rather than replace() based parse just for perf.